sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/datastore/V1Beta3.java

```diff
@@ -531,11 +531,17 @@ private RunQueryRequest makeRequest(Query query) {
 
     /**
      * Datastore system tables with statistics are periodically updated. This method fetches
-     * the latest timestamp of statistics update using the {@code __Stat_Total__} table.
+     * the latest timestamp of statistics update using the {@code __Stat_Total__} or
+     * {@code __Stat_Ns_Total__} table.
      */
     private long queryLatestStatisticsTimestamp(Datastore datastore) throws DatastoreException {
       Query.Builder query = Query.newBuilder();
-      query.addKindBuilder().setName("__Stat_Total__");
```

Comment:
> When the statistics system creates new statistic entities, it does not delete the old ones right away. The best way to get a consistent view of the statistics is to query for the __Stat_Total__ entity with the most recent timestamp, then use that timestamp value as a filter when fetching other statistic entities.

Writing down for posterity:
* Should we just use `__Stat_Total__`?

    I guess if we don't use multiple queries, we don't need a consistent view . So just use the namespace-specific variants.

Agree?